tools/qc/fastq_pair/fastq_pair.xml

```diff
@@ -0,0 +1,120 @@
+<tool id="fastq_pair" name="Paired-end fastq pairer" profile="18.01" version="1.0+galaxy0">
+    <description>Rewrite paired-end fastq files to make sure that all reads have a mate and to separate out singletons.</description>
+    <requirements>
+        <requirement type="package" version="1.0">fastq-pair</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+
+    ## Unzip gzipped inputs
+    #set fastq1_lw = str($fastq1).lower()
+    #set fastq2_lw = str($fastq2).lower()
+
+    #if fastq1_lw.endswith((".gz", ".gzip")):
```

Comment:
As you've discovered, that doesn't work because Galaxy (not planemo) does not store files under their uploaded names.
Data objects provide two things that will let you solve this:
- a `.is_of_type()` method, which you can use to check whether you're input param is of a given type known to Galaxy
- a `.ext` attribute, which is the file extension normally used with data of this type

Reply:
Oh wow thanks!